src/jemalloc.c

```diff
@@ -2925,15 +2932,6 @@ bool free_fastpath(void *ptr, size_t size, bool size_hint) {
 		}
 		assert(alloc_ctx.szind != SC_NSIZES);
 	} else {
-		/*
-		 * The size hinted fastpath does not involve rtree lookup, thus
-		 * can tolerate an uninitialized tsd.  This allows the tsd_fast
```

Comment:
In other words, this PR makes the free fastpath tolerate unintialized / nonfast tsd, so that it can avoid the tsd_fast() check the same way, using the next_event branch.